grakn-graql/src/main/java/ai/grakn/graql/internal/hal/HALConceptData.java

```diff
@@ -248,31 +252,48 @@ private void generateEntityEmbedded(Representation halResource, Entity entity, i
     }
 
     private void attachRelation(Representation halResource, Concept rel, String role, int separationDegree) {
-        Representation relationResource = factory.newRepresentation(resourceLinkPrefix + rel.getId()+this.keyspace)
+        Representation relationResource = factory.newRepresentation(resourceLinkPrefix + rel.getId() + this.keyspace)
                 .withProperty(DIRECTION_PROPERTY, INBOUND_EDGE);
         handleConcept(relationResource, rel, separationDegree - 1);
         halResource.withRepresentation(role, relationResource);
     }
 
 
     private void generateRelationEmbedded(Representation halResource, Relation rel, int separationDegree) {
+
+
         rel.rolePlayers().forEach((roleType, instance) -> {
             if (instance != null) {
-                Representation roleResource = factory.newRepresentation(resourceLinkPrefix + instance.getId()+this.keyspace)
+                Representation roleResource = factory.newRepresentation(resourceLinkPrefix + instance.getId() + this.keyspace)
                         .withProperty(DIRECTION_PROPERTY, OUTBOUND_EDGE);
                 handleConcept(roleResource, instance, separationDegree - 1);
                 halResource.withRepresentation(roleType.getName(), roleResource);
             }
         });
     }
 
+    private void embedRelationsPlaysRole(Representation halResource,Relation rel , int separationDegree){
```

Comment:
Is there a reason this doesn't use the existing behaviour in `generateEntityEmbedded`? I can't tell the difference.